### PR TITLE
smbus: Correct way to allow write block data to start with 0x.

### PR DESCRIPTION
**Pull request by jmeurin**
Signed-off-by: Jean-Marc Eurin <jmeurin@google.com>

### CODE REVIEW

File: smbus_rw.c [hunk 1]
```diff
@@ -313,23 +313,28 @@ parse_io_width(const char *arg, struct smbus_op_params *params,
 		char str_nibble[3];
 
 		len = strlen(arg);
+		/* Allow data starting with 0x */
+		if (arg[0] == '0' && (arg[1] == 'x' || arg[1] == 'X')) {
```
Comment (adurbin) [inline smbus_rw.c]: You shouldn't be reading into arg unless len is already at least 2.

File: smbus_rw.c [hunk 2]
```diff
@@ -313,23 +313,28 @@ parse_io_width(const char *arg, struct smbus_op_params *params,
...
+			arg += 2;
+			len -= 2;
+		}
+
 		if ( (len <= 0) || (len > 64) || (len % 2 != 0) ) {
 			fprintf(stderr, "%d: length is 0 or >64 or odd\n", len);
 			return -1;
 		}
+		params->len = len / 2;
```
Comment (adurbin) [inline smbus_rw.c]: Why is parens->len being manipulated at all here?

**Comment by adurbin** [comment]
In your commit description you aren't correcting anything. You are adding support for 0x prefix.

**Comment by jmeurin** [comment]
I hate git and github. 
I'm not correcting, I'm saying this is the correct way.
I move the params->len because 1) it's not used if the rest fails, 2) it keeps the len manipulations together and 3) I can now loop on the destination instead of the origin.

**Pull request closed without merging**